pkg/chunk/chunk_store.go

```diff
@@ -151,6 +151,12 @@ func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch
 
 // Get implements ChunkStore
 func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]local.SeriesIterator, error) {
+	now := model.Now()
+	if through.After(now) {
+		util.WithContext(ctx).Warnf("Capping TimeSpanHi from future=%v to now=%v", through, now)
+		through = now
```

Comment:
If `from` is also after `now` and this sets `through` to a value before `from`, we would return the `invalid query` error below. 

This may be confusing to the user (as we changed the `through` value internally), so perhaps we could handle this case separately and return no values?